crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -340,7 +341,9 @@ pub(crate) async fn analyze_ecmascript_module(
                 import_references.push(r);
             }
 
-            for r in import_references.iter_mut() {
+            // Avoid adding duplicate references to the analysis
+            // (this can work on unresolved Vcs as we can deduplicate on operations)
+            for r in import_references.iter_mut().collect::<IndexSet<_>>() {
                 // Resolving these references here avoids many resolve wrapper tasks when
                 // passing that to other turbo tasks functions later.
                 *r = r.resolve().await?;
```

Comment:
Oh actually this has a little bug with the deduplication, since it won't resolve all references. Could you change it to:

``` rs
            for r in import_references.iter_mut() {
                // Resolving these references here avoids many resolve wrapper tasks when
                // passing that to other turbo tasks functions later.
                *r = r.resolve().await?;
            }
            // Avoid adding duplicate references to the analysis
            for r in import_references.iter().collect::<IndexSet<_>>() {
                analysis.add_reference(*r);
            }
```